### PR TITLE
Hclang 23

### PR DESCRIPTION
**Pull request by jones1994**
### Description
Educational use of the arduino platform has been finalized.
Student and teacher interview has been finalized,

### Changelog
Changed educational section and added material to appendix.

### Issues solved 
add these with the respective key as HCLANG-1 (as an example)
 - HCLANG-23

## FOR REVIEWER [Do not remove]
A small checklist of things to note when you are doing a review. Make sure these things apply before going into master.
 - Did the PR actually fix the issue
 - Comment anything that doesn't make sense or anything you want clarified. Many comments are always ok
 - Make sure this is ready for exams if this is going into master
 - It's always a good idea having another reviewer look at a PR as well, if it's major. get a buddy.

### Regarding filenames
- all .tex files use pascal case
- all other files [except in /code/] use snake case

### Regarding code [if applicable]
- Check [coding conventions](https://kotlinlang.org/docs/reference/coding-conventions.html)
- Are public methods documented? 
- Are there testcases for new features / methods?
- Have you tried running the source and tested things that cannot be test-cased?

### Regarding Report [if applicable]
- Does each line have a linechange in the sourcecode?
- Does sentences make sense?
- Did you do spellchecking?
- Citations?

### CODE REVIEW

File: report/.gitignore [hunk 1]
```diff
@@ -1,3 +1,5 @@
+main.pdf
```
Comment (nillas12) [inline report/.gitignore]: Normalt ville man tilføje dette i en separat pull request (det er mere meta ting), men lad gå 🗡

Comment (jones1994) [reply report/.gitignore]: Den havde jeg helt glemt var med i den her pr

File: report/2.Analysis/ArduinoInEducation.tex [hunk 2]
```diff
@@ -0,0 +1,50 @@
+% point of document
+% We want to explain to the reader that (and how) the arduino platform is used in HTX
+% The point is:
+% - What "teknikfag" is
+% - What "el-teknik" is
+% 
+
+\section{Arduino and their usage in danish education}
+During the third year of the HTX\footnote{Higher Technical Examination Programme, is a 3-year vocationally oriented general upper secondary programme which builds on the 10th-11th form of the Folkeskole\cite{htx_wiki}} programme students are required to take a course in a voluntary technical subject, which utilizes a project-based learning practive. 
+There are a multiple of courses, one of these regard electronics, where many schools choose to utilize the Arduino platform\footnote{based on the fact that 4 of the students in this group studied at HTX in different parts of Denmark and all utilized the Arduino platform in this course}\cite{holstebro_education}.
```
Comment (nillas12) [inline report/2.Analysis/ArduinoInEducation.tex]: To the footnote, maybe add that the first school we contacted for questions also used arduino. Also good source.

File: report/2.Analysis/ArduinoInEducation.tex [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+% The point is:
+% - What "teknikfag" is
+% - What "el-teknik" is
+% 
+
+\section{Arduino and their usage in danish education}
+During the third year of the HTX\footnote{Higher Technical Examination Programme, is a 3-year vocationally oriented general upper secondary programme which builds on the 10th-11th form of the Folkeskole\cite{htx_wiki}} programme students are required to take a course in a voluntary technical subject, which utilizes a project-based learning practive. 
+There are a multiple of courses, one of these regard electronics, where many schools choose to utilize the Arduino platform\footnote{based on the fact that 4 of the students in this group studied at HTX in different parts of Denmark and all utilized the Arduino platform in this course}\cite{holstebro_education}.  
+
+This technical oriented course, often called electronic\footnote{The name varies between institutions}, aims to familiarize the students with electrical components and circuits as well as basic programming.\cite{holstebro_electronic}
```
Comment (nillas12) [inline report/2.Analysis/ArduinoInEducation.tex]: Change technical to technically. Change electronic to electronics (Pretty sure it should be plural)

File: report/2.Analysis/ArduinoInEducation.tex [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+Arduino is an ideal platform for this, as it is possible to connect I/O devices to the Arduino micro-controller, which in turn can be programmed to do simple tasks. 
+This course is then used to both spike the interest of the student regarding engineering fields, such as programming or electronics.
```
Comment (nillas12) [inline report/2.Analysis/ArduinoInEducation.tex]: Remove 'both', we only mention one thing (spiking an interest)

File: report/2.Analysis/ArduinoInEducation.tex [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+\subsection{HTX CAO Electronic Course}
+In relation to the discipline of understanding the initial problem. The project group contacted one of the major HTX institutions in North Jutland, Denmark. Namely HTX CAO. The following subsection will summarize the important and noticable aspects of the electronic course conducted on HTX CAO. The conclusions reached in this subsection are purely based on the statements made by the students and the teacher of this specific electronics course.
```
Comment (nillas12) [inline report/2.Analysis/ArduinoInEducation.tex]: Be better at new lining for one. First dot, should be a comma. Dot after Denmark, should be comma. Noticable should be notable. Do we mention what CAO means?

File: report/2.Analysis/ArduinoInEducation.tex [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+The electronics course itself does not teach the students programming directly. Instead it focuses on project based arduino development. All educational programming experience of the students (if any), originates from another optional course called "Programming". The Programming course teaches the students the basic contructions and termonology of general programming. it utilizes the languages C{\#}.
```
Comment (nillas12) [inline report/2.Analysis/ArduinoInEducation.tex]: It should be Capitalized. Change utilized to: It is taught in/with the language C#. (Be sure to remove 's' from languages)

File: report/2.Analysis/ArduinoInEducation.tex [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+The electronics course itself does not teach the students programming directly. Instead it focuses on project based arduino development. All educational programming experience of the students (if any), originates from another optional course called "Programming". The Programming course teaches the students the basic contructions and termonology of general programming. it utilizes the languages C{\#}. 
+
+The electronics course focuses on engaging the students in making arudino projects, not in making the students understand the theory behind the projects. This means that the students both have a very limited understanding of the arudino platform's architecture and a low understanding of the arduino language explained earlier in the report, which they utilize in the electronics course on HTX CAO.
```
Comment (nillas12) [inline report/2.Analysis/ArduinoInEducation.tex]: change: ", not in making the students understand the theory behind the projects" to "not teaching the theory behind the  projects" (or something like that).
Change arudino to arduino (spelling error).
Change explained to as explained. Also include a reference to where this was explained.

File: report/2.Analysis/ArduinoInEducation.tex [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+
+Because of all this the electronics course have a relatively high learning curve. Which results in a high number of students abandoning the course in favor of some of the courses with a more specificly defined theory structures.
```
Comment (nillas12) [inline report/2.Analysis/ArduinoInEducation.tex]: Remove all. Also "have a relatively high learning curve" to "has a relatively high learning curve".
Also last part needs to be reformulated to: "In favor of courses with a more defined structure" (optionally add theory)

File: report/2.Analysis/ArduinoInEducation.tex [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+
+Clearly there is a majojr gap between the students who are interested in programming or electronics and have no prior experience and students who have prior experiences.
```
Comment (nillas12) [inline report/2.Analysis/ArduinoInEducation.tex]: Majojr to major. change after and to: "And the students who have prior experience"

File: report/2.Analysis/ArduinoInEducation.tex [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+
+The electronics course focuses on engaging the students in making arudino projects, not in making the students understand the theory behind the projects. This means that the students both have a very limited understanding of the arudino platform's architecture and a low understanding of the arduino language explained earlier in the report, which they utilize in the electronics course on HTX CAO.
+
+Because of all this the electronics course have a relatively high learning curve. Which results in a high number of students abandoning the course in favor of some of the courses with a more specificly defined theory structures.
+
+Clearly there is a majojr gap between the students who are interested in programming or electronics and have no prior experience and students who have prior experiences. 
+Based on students' and the teacher's statements it seems like the course needs a better gateway into the arduino platform, the programming aspects of the course or both.
```
Comment (nillas12) [inline report/2.Analysis/ArduinoInEducation.tex]: Change: "Based on student's" to "based on the students' ". Add comma after "teacher's statements". Remove 's' from 'needs'.

File: report/2.Analysis/ArduinoInEducation.tex [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+
+A substantial number of the students in the class expressed their frustation with relative complexity of the arduino language. The teacher supplemented with saying that the language is not necessary complex, but the time spent on the different concepts are far too short, which is a result of the education being a junior education and not a college/university level education.
```
Comment (nillas12) [inline report/2.Analysis/ArduinoInEducation.tex]: Change "with relative complexity" to "with the relative complexity". Change "necessary" to "necessarily". Change "concepts are far too short" to "concepts is far too short"

File: report/2.Analysis/ArduinoInEducation.tex [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+Clearly there is a need for the course to be able to make shortcuts in some aspects to teach the students the concepts adequately. Put simply the language and the platform are not well suited for this kind of low level education.
```
Comment (nillas12) [inline report/2.Analysis/ArduinoInEducation.tex]: "Put simply" to "Simply put," (remember comma). Change "platform are not" to "platform is not"

File: report/2.Analysis/ArduinoInEducation.tex [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+
+\subsubsection{The Students Understanding of Syntax and Semantics}
+Most of the students expressed that they have no real interest in the syntax. In fact a substantial part indicated that they did not understand what the syntax of a programming language actually means. Below is a short summarization of the most problematic aspects of general syntax and semantic of the arduino language (or any reference language the students might have), in arbitary order.
```
Comment (nillas12) [inline report/2.Analysis/ArduinoInEducation.tex]: Add s to semantic

File: report/2.Analysis/ArduinoInEducation.tex [hunk 14]
```diff
@@ -0,0 +1,50 @@
...
+
+\textbf{Data Types}\\
+Data types in general, seems to cause alot of problems for the students. Most students can name the different types but far less have the proper knowledge to identify and explain the meaning behind them. The students in most cases do not understand why they have to state the data type explicitly when declaring a variable. When presented with a small example program utilizing type inference on all declarations, the students in most cases, instantly understood how to identify a declaration and how to use the variable properly.
```
Comment (nillas12) [inline report/2.Analysis/ArduinoInEducation.tex]: Change "Data types in general, seems to cause alot of problems for the students." to "Data types in general, seem to cause problems for the students"

Comment (nillas12) [inline report/2.Analysis/ArduinoInEducation.tex]: Add a comma at: "state the data type expicitly, when declaring a variable" Change "utilizing" to "with" (simpler word)

File: report/2.Analysis/ArduinoInEducation.tex [hunk 15]
```diff
@@ -0,0 +1,50 @@
...
+
+\textbf{Function Call and Definition}\\
+The students concluded that the manner in which C++ and C{\#} makes function/method calls and definitions are somewhat ambigues to them. In relation to syntax they expressed some minor frustrations concerning the overhead nessacery to define simple methods, such as methods that does not return a value and also when making function calls with no arguments. The most problematic aspect of abstraction in general was the matter of scoping. No students expressed any real knowledge of scoping. This is especiallly a problem when it comes to understanding error messages from the compiler. Clearly scoping rules must be kept in mind when developing a new language for beginners.
```
Comment (nillas12) [inline report/2.Analysis/ArduinoInEducation.tex]: Change "calls and definitions are somewhat ambigues to them" to "calls and definitions is somewhat ambiguous to them". Add comma between "syntax" and "they"

File: report/2.Analysis/ArduinoInEducation.tex [hunk 16]
```diff
@@ -0,0 +1,50 @@
...
+
+\textbf{Control Structures}\\
+In general the students understood what the different controlstructures do. But almost non of them could remember the specific syntax and meaning behind them. The selective control structures seems to be easier to understand than the iterative control structures. A substantial amount of the students stated that the control structes in general are difficult to understand and remember, since they do not reflect anything else in the language. At least according to them.
```
Comment (nillas12) [inline report/2.Analysis/ArduinoInEducation.tex]: Change "language. At least according to them." to "language, at least according to them."

File: report/2.Analysis/ArduinoInEducation.tex [hunk 17]
```diff
@@ -0,0 +1,50 @@
...
+Clearly the students needs more consistensy in the language. The control structures should in some way mirror the syntax of another major aspect of the languange, to make it more intuitive to use and remember.
```
Comment (nillas12) [inline report/2.Analysis/ArduinoInEducation.tex]: "consistency" to "consitency"

File: report/2.Analysis/ArduinoInEducation.tex [hunk 18]
```diff
@@ -0,0 +1,50 @@
...
+
+\textbf{Operator Precedence and Association}\\
+Both terms were completely forign to the students. Their perception of precedence were mostly based on arithmetic notation. This is sufficient in most cases. The biggest concern was with association. Most of students did not understand that different operators could have different association. which proved to be a major issue when presented with code. In most cases the students understood what part of the expression was calculated first. But choose the wrong anwser based on mostly random factors. Clearly association needs to be as intuitive as possible.
```
Comment (nillas12) [inline report/2.Analysis/ArduinoInEducation.tex]: Change "but choose the wrong anwser" to "but chose the wrong answer" (Both spelling and an o too much)

File: report/2.Analysis/ArduinoInEducation.tex [hunk 19]
```diff
@@ -0,0 +1,50 @@
...
+
+\subsection{Summary}
+Based on interviews done with students having this course, the language used for Arduino, C++, has a steep learning curve, considering the experience possesed by the students. The Arduino, from a hardware point of view, is quite intuitive and it is easy to play around with lights and similar things that give the student a visual feedback. Even though the students have no actual understanding of the architecture of the platform they essentially do not need it. Many students interviewed felt like the programming aspects of the course are far to dificult to understand in its entirety (which is reasonable), but even simple syntactic understanding og the programming language are difficult to understand for most students. The notes concerning the inteviews analized above can be found in the Appendix 1 on page \pageref{Interviews}.
```
Comment (nillas12) [inline report/2.Analysis/ArduinoInEducation.tex]: Remove "Things". Maybe I/O, Modules, components?

File: report/2.Analysis/ArduinoInEducation.tex (hunk 5 above)
Comment (jones1994) [reply report/2.Analysis/ArduinoInEducation.tex]: Well CAO is a name

File: report/2.Analysis/ArduinoInEducation.tex (hunk 9 above)
Comment (jones1994) [reply report/2.Analysis/ArduinoInEducation.tex]: I do not understand the second part of this comment

**Pull request merged**